components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -1181,7 +1184,12 @@ private String getBlobValue(InputStream is) throws IdentityProviderManagementExc
     public IdentityProvider getIdPByName(Connection dbConnection, String idPName, int tenantId,
                                          String tenantDomain) throws IdentityProviderManagementException {
 
-        return getIDP(dbConnection, idPName, -1, tenantId, tenantDomain);
+        ExtendedIdentityProvider extendedIdp = getIDP(dbConnection, idPName, -1, null, tenantId, tenantDomain);
```

Comment:
Check the reason for accepting a connection object to this method. If there is no particular requirement, please remove it.

Reply:
in addIDP, updateIDP, this method is called with db connection to retrieve the created/ updated IDP object. Also deleteIDP invokes this method to check whether resource exists.